website/blog/2023-06-01-ariadne-codegen-0-7.md

```diff
@@ -0,0 +1,307 @@
+---
+title: Ariadne Codegen 0.7
+---
+
+Ariadne Codegen 0.7 is now available!
+
+This release brings support for subscriptions, changes how fragments are represented in generated code, introduces `ShorterResultsPlugin` plugin developed by our amazing community and more features and fixes.
+
+<!--truncate-->
+
+## Subscriptions
+
+Version `0.7` introduces support for subscriptions. We generate them as async generators, which means that we don't support subscriptions when generated client is not async (`async_client` is set to `false`).
+
+For example, given following operation:
+
+```gql
+subscription GetUsersCounter {
+    usersCounter
+}
+```
+
+Generated client will have following method:
+
+```py
+    async def get_users_counter(self) -> AsyncIterator[GetUsersCounter]:
+        query = gql(
+            """
+            subscription GetUsersCounter {
+              usersCounter
+            }
+            """
+        )
+        variables: dict[str, object] = {}
+        async for data in self.execute_ws(query=query, variables=variables):
+            yield GetUsersCounter.parse_obj(data)
+```
+
+Our default async base client uses [websockets](https://github.com/python-websockets/websockets) package and implements [graphql-transport-ws](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md) subprotocol.
+
+Required dependencies can by installed with pip:
+
+```
+$ pip install ariadne-codegen[subscriptions]
+```
+
+
+## Fragments
+
+In previous versions of Codegen fragments were "unpacked" in queries. For example, given following operations:
+
+```gql
+query GetA {
+    GetTypeA {
+        ...FragmentA
+    }
+}
+
+query ListA {
+    ListTypeA {
+        ...FragmentA
+    }
+}
+
+fragment FragmentA on TypeA {
+    id
+    name
+}
+```
+
+Generated `get_a.py` and `list_a.py` files had types looking like this:
+
+```py
+# get_a.py
+
+class GetA(BaseModel):
+    get_type_a: "GetAGetTypeA" = Field(alias="GetTypeA")
+
+
+class GetAGetTypeA(BaseModel):
+    id: str
+    name: str
+```
+
+```py
+# list_a.py
+
+class ListA(BaseModel):
+    list_type_a: List["ListAListTypeA"] = Field(alias="ListTypeA")
+
+
+class ListAListTypeA(BaseModel):
+    id: str
+    name: str
+```
+
+Both of these operations use the same `FragmentA` to represnt `TypeA`, but our models didn't show that.
+
+To make working with fragment easier, in Ariadne Codegen 0.7 we are changing this behaviour. Instead of unpacking fragments, we generate separate models for them and use them as mixins. From the same operation now we will get 3 files: `get_a.py`, `list_a.py` and `fragments.py`
```

Comment:
```suggestion
To make working with fragments easier, in Ariadne Codegen 0.7 we are changing this behavior. Instead of unpacking fragments, we generate separate models from them and use those as mixins. Above operation will now result in 3 files being generated: `get_a.py`, `list_a.py` and `fragments.py`
```